x-pack/plugins/actions/server/integration_tests/axios_utils_connection.test.ts

```diff
@@ -208,6 +210,72 @@ describe('axios connections', () => {
       const fn = async () => await request({ axios, url, logger, configurationUtilities });
       await expect(fn()).rejects.toThrow('certificate');
     });
+
+    test('it works with ca in SSL overrides', async () => {
+      const { url, server } = await createServer({ useHttps: true });
+      testServer = server;
+
+      const configurationUtilities = getACUfromConfig();
+      const sslOverrides = {
+        ca: Buffer.from(CA),
+      };
+      const res = await request({ axios, url, logger, configurationUtilities, sslOverrides });
+      expect(res.status).toBe(200);
+    });
+
+    test('it works with cert, key, and ca in SSL overrides', async () => {
+      const { url, server } = await createServer({ useHttps: true });
```

Comment:
I think this isn't going to test the client-side certificate stuff.  We'll need to add another option to `createServer()` to have it pass [`requestCert`](https://nodejs.org/dist/latest-v18.x/docs/api/tls.html#tlscreateserveroptions-secureconnectionlistener) in the TLS options when creating the server.  Otherwise, the server is just going to ignore those.

To make sure this is working, we should have a negative test as well.  Make a request to a client-side enabled server WITHOUT any of the client-side stuff, and it should fail.  If we had that test in now, presumably the request would succeed, so the test would fail.

Reply:
Adding `requestCert` and getting `socket hang up` errors. This seems like what Node does when you're sending it an invalid client certificate but I'm not sure, docs are very unclear. I'm still investigating.

Reply:
Figured it out, server wasn't being configured with the Kibana CA by default. Tests are pushed and working now.